### PR TITLE
[Spotless] Applying Google Code Format for entire project #22

### PR DESCRIPTION
**Pull request by MitchellGale**
### Description
Applies spotless on entire project with `**/*.java`
Removes checkstyles from project
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/1101
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Yury-Fridlyand — COMMENTED** [review]
Can you move left of changes in `:legacy` into another PR?

File: .github/workflows/sql-test-workflow.yml [hunk 1]
```diff
@@ -31,6 +31,16 @@ jobs:
     - name: Run tests
       id: tests
       run: |
+        # Spotless
+        ./gradlew :opensearch:spotlessCheck              || echo "* Spotless failed for opensearch/src" > report.log
+        ./gradlew :sql:spotlessCheck                     || echo "* Spotless failed for sql/src" >> report.log
+        ./gradlew :ppl:spotlessCheck                     || echo "* Spotless failed for ppl/src" >> report.log
+        ./gradlew :core:spotlessCheck                    || echo "* Spotless failed for core/test" >> report.log
```
Comment (Yury-Fridlyand) [inline .github/workflows/sql-test-workflow.yml:38]: Is `spotlessCheck` runs for src or for test or for both?

Comment (MitchellGale) [reply .github/workflows/sql-test-workflow.yml]: I need to add it to each of their `build.gradle` for it to work

Comment (MitchellGale) [reply .github/workflows/sql-test-workflow.yml]: but it'll be for both

**Review by acarbonetto — DISMISSED** [review]

File: legacy/src/main/java/org/opensearch/sql/legacy/rewriter/nestedfield/From.java [hunk 2]
```diff
@@ -70,29 +70,31 @@ private void keepParentTableOnly() {
...
-     */
-    private void collectNestedFields(Scope scope) {
-        From clause = this;
-        for (; clause.isCommaJoin(); clause = clause.right()) {
-            clause.left().addIfNestedField(scope);
-        }
-        clause.addIfNestedField(scope);
+  /**
+   *
+   *
```
Comment (matthewryanwells) [inline legacy/src/main/java/org/opensearch/sql/legacy/rewriter/nestedfield/From.java:75]: Looks like we have the same issue with 2 asterisks that Andrew pointed out on another spotless PR, did we figure out what caused this and how it was fixed?

File: legacy/src/main/java/org/opensearch/sql/legacy/rewriter/nestedfield/Identifier.java [hunk 3]
```diff
@@ -9,6 +9,8 @@
 import com.alibaba.druid.sql.ast.expr.SQLIdentifierExpr;
 
 /**
+ *
+ *
```
Comment (matthewryanwells) [inline legacy/src/main/java/org/opensearch/sql/legacy/rewriter/nestedfield/Identifier.java:13]: Same thing here

File: common/build.gradle [hunk 4]
```diff
@@ -68,3 +64,21 @@ configurations.all {
...
 }
+
+spotless {
+    java {
+        target fileTree('.') {
+            include '**/*.java'
+            exclude '**/build/**', '**/build-*/**'
+        }
+        importOrder()
+//        licenseHeader("/*\n" +
```
Comment (acarbonetto) [inline common/build.gradle:75]: if we aren't going to apply this, we should either have a comment or remove it.

**Review by acarbonetto — APPROVED** [review]

File: common/build.gradle (hunk 4 above)
Comment (MitchellGale) [inline common/build.gradle:75]:
```suggestion
// Needs https://github.com/opensearch-project/sql/issues/1893 to be addressed first
//        licenseHeader("/*\n" +
```

Comment (MitchellGale) [reply common/build.gradle:76]: Good idea, just added a comment that references the blocking issue

**Review by matthewryanwells — APPROVED** [review]

File: integ-test/build.gradle [hunk 5]
```diff
@@ -529,6 +524,24 @@ List<Provider<RegularFile>> plugins = [
     })
 ]
 
+spotless {
```
Comment (Yury-Fridlyand) [inline integ-test/build.gradle:527]: You have spotless here (in this file) twice, this causes build to fail

File: integ-test/build.gradle [hunk 6]
```diff
@@ -529,6 +524,24 @@ List<Provider<RegularFile>> plugins = [
...
+//                " * Copyright OpenSearch Contributors\n" +
+//                " * SPDX-License-Identifier: Apache-2.0\n" +
+//                " */\n\n")
+        removeUnusedImports()
+        trimTrailingWhitespace()
+        endWithNewline()
+        googleJavaFormat('1.17.0').reflowLongStrings().groupArtifact('com.google.googlejavaformat:google-java-format')
+    }
+}
+
```
Comment (Yury-Fridlyand) [inline integ-test/build.gradle:544]:
```suggestion
```

**Review by acarbonetto — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Comment by joshuali925** [comment]
removing backport-failed since [there's a PR](https://github.com/opensearch-project/sql/pull/1994) merged for 2.x

**Pull request merged**